src/Symfony/Component/Notifier/Bridge/GoogleChat/GoogleChatTransportFactory.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\GoogleChat;
+
+use Symfony\Component\Notifier\Exception\UnsupportedSchemeException;
+use Symfony\Component\Notifier\Transport\AbstractTransportFactory;
+use Symfony\Component\Notifier\Transport\Dsn;
+use Symfony\Component\Notifier\Transport\TransportInterface;
+
+/**
+ * @author Jérôme Tamarelle <jerome@tamarelle.net>
+ *
+ * @experimental in 5.1
+ */
+final class GoogleChatTransportFactory extends AbstractTransportFactory
+{
+    /**
+     * @param Dsn $dsn Example: googlechat://chat.googleapis.com/v1/spaces/<space>/messages?key=<accessKey>&token=<accessToken>
+     */
+    public function create(Dsn $dsn): TransportInterface
+    {
+        $scheme = $dsn->getScheme();
+
+        if ('googlechat' === $scheme) {
+            $space = explode('/', $dsn->getPath())[3];
+            $accessKey = $dsn->getOption('key');
+            $accessToken = $dsn->getOption('token');
```

Comment:
You should probably throw an exception of the key and/or token is empty. Or perhaps better, user the username/password like other providers.

Reply:
I like the simplicity of copy-paste the webhook url without too much alteration. Changing the scheme is already a trap I fell into while coding this bridge.

But maybe it's a good thing to move apart from the url provided by Google in order to make clear for developers that they have to parse it in their mind and rewrite it following a custom pattern.

Reply:
Regarding how other DSN are made, I chose to use a DSN totally different from the webhook url.

Reply:
After a review of [existing DSN](https://symfony.com/doc/current/notifier.html#chat-channel), I applied the following format:
```
googlechat://ACCESS_KEY:ACCESS_TOKEN@default/SPACE?threadKey=THREAD
```
Where `THREAD` is optional.